qintelqwatch_connector.py

```diff
@@ -0,0 +1,412 @@
+            if ts['context'] == 'loaded':
+                ret_ts['loaded'] = f'{ts["iso"]}Z'
+
+        # first seen - cred
+        timestamps = exposure['meta']['stats']['credential']['timestamps']
+        for ts in timestamps:
+            if ts['context'] == 'first_seen':
+                ret_ts['first_seen'] = f'{ts["iso"]}Z'
+
+        return ret_ts
+
+    def _process_qwatch_exposure(self, exposure):
+
+        return_data = {}
+
+        timestamps = self._process_timestamps(exposure)
+
+        return_data['credential'] = exposure['attributes']['login_name']
+
+        return_data['password'] = None
+        if self.plaintext_passwords:
+            return_data['password'] = exposure['attributes']['password']
+
+        return_data['source_name'] = exposure['attributes']['source_name']
+        return_data['last_seen'] = timestamps.get('loaded')
+        return_data['first_seen'] = timestamps.get('first_seen')
+
+        return return_data
+
+    def _set_severity(self, timestamps, cef):
+
+        source_name = cef['exposureSourceName']
+        if source_name.lower().startswith('malware'):
+            return 'high'
+
+        if timestamps['loaded'] == timestamps['first_seen']:
+            return 'high'
+
+        return 'medium'
+
+    def _process_exposures_ingest(self, data):
+
+        total_creds = data['meta']['total']
+        self.save_progress(f'Processing {total_creds} exposures')
+
+        if total_creds == 0:
+            return
+
+        container = {
+            'name': 'Qintel QWatch Alert',
+            'description': f'{total_creds} exposed credentials',
+            'artifacts': []
+        }
+
+        loaded_ts = []
+
+        for exposure in data.get('data', []):
+            artifact = {}
+
+            timestamps = self._process_timestamps(exposure)
+            loaded_ts.append(timestamps.get('loaded'))
+
+            cef = {}
+            cef_types = {}
+
+            cef['exposedCredential'] = exposure['attributes']['login_name']
+            cef_types['exposedCredential'] = ['qwatch credential', 'email']
+
+            cef['exposureSourceName'] = exposure['attributes']['source_name']
+            cef_types['exposureSourceName'] = ['qwatch source name']
+
+            artifact['start_time'] = timestamps.get('first_seen')
+            artifact['end_time'] = timestamps.get('loaded')
+
+            artifact['severity'] = self._set_severity(timestamps, cef)
+            artifact['label'] = 'event'
+            artifact['name'] = cef['exposedCredential']
+
+            artifact['cef'] = cef
+            artifact['cef_types'] = cef_types
+            artifact['data'] = exposure
+
+            container['artifacts'].append(artifact)
+
+        status, msg, id_ = self.save_container(container)
+        if status == phantom.APP_ERROR:
+            raise Exception(f'Container creation failed: {msg}')
+
+        # save poll time
+        self._state['last_poll'] = sorted(loaded_ts)[-1]
```

Comment:
Could we also mention how poll time will be used? I assume it will be used to avoid duplicate polling.